### PR TITLE
Search page

### PR DESCRIPTION
**Pull request by macebm**
Fixes from last PR and some style changes. 

Now new page Search page. You can click on a category and get all products with it

### CODE REVIEW

File: frontend/src/components/GlobalStyle.js [hunk 1]
```diff
@@ -8,7 +7,8 @@ export default createGlobalStyle`
         box-sizing: border-box;
     }
     body{
-        background: ${Colors.secondaryBackground};
+
+        background: url("https://cdn.pixabay.com/photo/2019/05/06/19/13/green-4183977_1280.jpg");
```
Comment (web-devil) [inline frontend/src/components/GlobalStyle.js]: lieber im repository hinterlegen

File: frontend/src/components/Search.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+export default function Search() {
+  const [products, setProducts] = useState([]);
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    getCategories()
+      .then((data) => setCategories(data))
+      .catch((error) => console.error(error.message));
+  }, []);
+  function handleCategory(categoryId) {
```
Comment (web-devil) [inline frontend/src/components/Search.js:20]: leerzeile

**Review by web-devil — CHANGES_REQUESTED** [review]

File: frontend/src/components/CategoryButton.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import styled from "styled-components/macro";
+import Colors from "../constants/Colors";
+
+export default function CategoryButton({ id, children, onClick }) {
+  return <Wrapper onClick={() => onClick(id)}>{children}</Wrapper>;
```
Comment (ThomasKirst) [inline frontend/src/components/CategoryButton.js]:
```suggestion
  return <Button onClick={() => onClick(id)}>{children}</Button>;
```
`Wrapper` is not a great component name, you could rather use `Button`, for instance.

File: frontend/src/components/CategoryButton.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import styled from "styled-components/macro";
+import Colors from "../constants/Colors";
```
Comment (ThomasKirst) [inline frontend/src/components/CategoryButton.js:2]: Reconsider fully capitalised letters as constants should be declared as such.

File: frontend/src/components/ProductWithStore.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+export default function ProductWithStore({
+  manufacturer,
+  name,
+  price,
+  store,
+  postalCode,
+  street,
+}) {
+  return (
+    <Wrapper>
```
Comment (ThomasKirst) [inline frontend/src/components/ProductWithStore.js]: Another `Wrapper` – why not `Product` or `ProductCard`?

File: frontend/src/components/Search.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    getProductsByCategory(categoryId)
+      .then((data) => setProducts(data))
+      .catch((error) => console.error(error));
+  }
+
+  return (
+    <Wrapper>
```
Comment (ThomasKirst) [inline frontend/src/components/Search.js]: Is this a `<SearchPage…` or really another `Wrapper`?

File: frontend/src/components/Search.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  color: ${Colors.headers};
+  text-align: center;
+`;
+
+const StyledSection = styled.section`
+  display: flex;
+  flex-wrap: wrap;
+  gap: 2rem;
+  justify-content: space-evenly;
+  margin-bottom: 3.75rem;
```
Comment (ThomasKirst) [inline frontend/src/components/Search.js:73]: `rem`s to the rescue! We like 👍

File: frontend/src/components/SingleStore.js [hunk 8]
```diff
@@ -44,14 +45,18 @@ export default function SingleStore() {
       <h2>Milk</h2>
 
       <StyledProductBox>
```
Comment (ThomasKirst) [inline frontend/src/components/SingleStore.js]:
```suggestion
      <ProductBox>
```
For sure it's going to be styled ;)

File: frontend/src/components/SingleStore.js [hunk 9]
```diff
@@ -74,20 +79,25 @@ export default function SingleStore() {
...
+              name={name}
+              price={price}
+              manufacturer={manufacturer}
+            />
+          ))
+        ) : (
+          <h2>No Meat</h2>
+        )}
       </StyledProductBox>
     </StyledSection>
```
Comment (ThomasKirst) [inline frontend/src/components/SingleStore.js]:
```suggestion
    </Section>
```

**Review by ThomasKirst — COMMENTED** [review]
Overall very good! Just a few remarks about naming of components.

You can .gitignore this file: .DS_Store 

Great Job! Miriam says: 😸

File: frontend/src/components/Search.js (hunk 6 above)
Comment (macebm) [reply frontend/src/components/Search.js]: Its a page

File: frontend/src/components/CategoryButton.js (hunk 3 above)
Comment (macebm) [reply frontend/src/components/CategoryButton.js]: Changed it for all thank you 🍀

**Review by miriamhaenle — APPROVED** [review]

**Pull request merged**